client/src/ui/PanelCarousel.lua

```diff
@@ -22,7 +23,14 @@ function PanelCarousel:createPassenger(id)
   -- always add shock
   stackPanel:addElement(panelImages[8])
 
-  return {id = id, uiElement = stackPanel, panelImages = panelImages}
+  local overallStack = StackPanel({alignment = "top", hAlign = "center", vAlign = "center"})
+  local nameLabel = Label({text = panels[id].name or "", translate = false})
+  nameLabel.hAlign = "center"
+
+  overallStack:addElement(stackPanel)
+  overallStack:addElement(nameLabel)
+
+  return {id = id, uiElement = overallStack, panelImages = panelImages}
```

Comment:
Should probably revert this one.
It looks broken in 2p rooms and is more of an afterthought to your actual intentions for this PR so it shouldn't hold it up. We should bring that back when reorganizing room real estate.